commit d29be158a68254f58cf1fbf60ce1e89557a321aa
Author: Miloslav Trmac <mitr@redhat.com>
Date:   Thu Sep 16 18:14:11 2010 -0400

    Audit: add support to match lsm labels on user audit messages
    
    Add support for matching by security label (e.g. SELinux context) of
    the sender of an user-space audit record.
    
    The audit filter code already allows user space to configure such
    filters, but they were ignored during evaluation.  This patch implements
    evaluation of these filters.
    
    For example, after application of this patch, PAM authentication logs
    caused by cron can be disabled using
            auditctl -a user,never -F subj_type=crond_t
    
    Signed-off-by: Miloslav Trmac <mitr@redhat.com>
    Acked-by: Eric Paris <eparis@redhat.com>
    Signed-off-by: Al Viro <viro@zeniv.linux.org.uk>

diff --git a/kernel/auditfilter.c b/kernel/auditfilter.c
index eb7675499fb5..add2819af71b 100644
--- a/kernel/auditfilter.c
+++ b/kernel/auditfilter.c
@@ -1252,6 +1252,18 @@ static int audit_filter_user_rules(struct netlink_skb_parms *cb,
 		case AUDIT_LOGINUID:
 			result = audit_comparator(cb->loginuid, f->op, f->val);
 			break;
+		case AUDIT_SUBJ_USER:
+		case AUDIT_SUBJ_ROLE:
+		case AUDIT_SUBJ_TYPE:
+		case AUDIT_SUBJ_SEN:
+		case AUDIT_SUBJ_CLR:
+			if (f->lsm_rule)
+				result = security_audit_rule_match(cb->sid,
+								   f->type,
+								   f->op,
+								   f->lsm_rule,
+								   NULL);
+			break;
 		}
 
 		if (!result)